src/server/game/AI/SmartScripts/SmartScriptMgr.cpp

```diff
@@ -75,39 +67,35 @@ void SmartWaypointMgr::LoadFromDB()
         y = fields[3].GetFloat();
         z = fields[4].GetFloat();
 
+        SmartWaypointPath& path = waypointMap[entry];
+
         if (last_entry != entry)
         {
-            waypoint_map[entry] = new WPPath();
             last_id = 1;
-            count++;
+            ++count;
         }
 
         if (last_id != id)
             TC_LOG_ERROR("sql.sql", "SmartWaypointMgr::LoadFromDB: Path entry %u, unexpected point id %u, expected %u.", entry, id, last_id);
 
-        last_id++;
-        (*waypoint_map[entry])[id] = new WayPoint(id, x, y, z);
+        ++last_id;
+
+        WayPoint point;
+        point.id = id;
+        point.x = x;
+        point.y = y;
+        point.z = z;
 
+        path.push_back(std::move(point));
```

Comment:
Why didn't you keep the old constructor accepting id, x, y and z as args and use `path.emplace_back(id, x, y, z)` here?